src/openzeppelin/tests/test_account.cairo

```diff
@@ -0,0 +1,330 @@
+use core::traits::Into;
+use array::ArrayTrait;
+use array::SpanTrait;
+use core::result::ResultTrait;
+use option::OptionTrait;
+use serde::Serde;
+use starknet::class_hash::Felt252TryIntoClassHash;
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::syscalls::deploy_syscall;
+use starknet::testing;
+use traits::TryInto;
+
+use openzeppelin::account::Account;
+use openzeppelin::account::IAccountDispatcher;
+use openzeppelin::account::IAccountDispatcherTrait;
+use openzeppelin::account::ERC165_ACCOUNT_ID;
+use openzeppelin::account::ERC1271_VALIDATED;
+use openzeppelin::account::TRANSACTION_VERSION;
+use openzeppelin::account::Call;
+use openzeppelin::token::erc20::ERC20;
+use openzeppelin::token::erc20::IERC20Dispatcher;
+use openzeppelin::token::erc20::IERC20DispatcherTrait;
+use openzeppelin::introspection::erc165::IERC165_ID;
+
+#[derive(Drop)]
+struct SignedTransactionData {
+    private_key: felt252,
+    public_key: felt252,
+    transaction_hash: felt252,
+    r: felt252,
+    s: felt252
+}
+
+fn PUBLIC_KEY() -> felt252 {
+    0x333333
+}
+fn ACCOUNT_ADDRESS() -> ContractAddress {
+    contract_address_const::<0x111111>()
+}
+fn SIGNED_TX_DATA() -> SignedTransactionData {
+    SignedTransactionData {
+        private_key: 1234,
+        public_key: 883045738439352841478194533192765345509759306772397516907181243450667673002,
+        transaction_hash: 2717105892474786771566982177444710571376803476229898722748888396642649184538,
+        r: 3068558690657879390136740086327753007413919701043650133111397282816679110801,
+        s: 3355728545224320878895493649495491771252432631648740019139167265522817576501
+    }
+}
+
+fn setup_dispatcher(data: Option<@SignedTransactionData>) -> IAccountDispatcher {
+    // Set the transaction version
+    testing::set_version(TRANSACTION_VERSION);
+
+    // Deploy the account contract
+    let mut calldata = ArrayTrait::<felt252>::new();
+
+    if data.is_some() {
+        let data = data.unwrap();
+
+        // Set the signature and transaction hash
+        let mut signature = ArrayTrait::new();
+        signature.append(*data.r);
+        signature.append(*data.s);
+        testing::set_signature(signature.span());
+        testing::set_transaction_hash(*data.transaction_hash);
+
+        calldata.append(*data.public_key);
+    } else {
+        calldata.append(PUBLIC_KEY());
+    }
+
+    let (address, _) = deploy_syscall(
+        Account::TEST_CLASS_HASH.try_into().unwrap(), 0, calldata.span(), false
```

Comment:
```suggestion

    let (address, _) = starknet::deploy_syscall(
        Account::TEST_CLASS_HASH.try_into().unwrap(), 0, calldata.span(), false
```

This is preferred from the [discussion](https://github.com/OpenZeppelin/cairo-contracts/pull/604#discussion_r1190914214), right?

Reply:
It is a Rust common pattern, that I don't think is the best option in this particular case. The pattern is using the module wrapping the functions instead of the functions directly. I proposed this for setters inside testing, like testing::set_version, because there are a lot of setters and the set_version name is not descriptive enough by itself IMO, but I agree deploy_syscall is not the same.